clientv3/balancer.go

```diff
@@ -64,6 +64,23 @@ func newSimpleBalancer(eps []string) *simpleBalancer {
 	return sb
 }
 
+func (b *simpleBalancer) update(m map[string]struct{}) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	addrs := make([]grpc.Address, 0, len(m))
+	for ep := range m {
+		addrs = append(addrs, grpc.Address{Addr: getHost(ep)})
+	}
+	b.addrs = addrs
+	b.notifyCh <- addrs
```

Comment:
I'm not really familiar with the semantics of the notify channel. I don't think this function should be resetting readyc/upEps/upc/pinAddr though; anything that's waiting on `upc` or `readyc` will break for sure.